src/models/db-d2.ts

```diff
@@ -275,27 +275,84 @@ export default class DbD2 {
+    ) {
+        const allAncestorsIds = _(organisationUnitsPathOnly)
+            .map("path")
+            .flatMap(path => path.split("/").slice(1))
+            .uniq()
+            .value()
+            .join(",");
+
+        const orgUnitsId = _(organisationUnitsPathOnly).map("id");
+        const dashboardItems = [dashboardItemsConfig.charts, dashboardItemsConfig.tables];
+        const elements = _(dashboardItems)
             .values()
-            .flatten();
+            .flatMap(_.values)
+            .groupBy("dataType")
+            .mapValues(objs => _.flatMap(objs, "elements"))
+            .value();
+
+        const allDataElementCodes = elements["DATA_ELEMENT"].join(",");
+        const allIndicatorCodes = elements["INDICATOR"].join(",");
         const antigenCodes = antigens.map(an => an.code);
-        const { dataElements, categories, indicators } = await this.api.get("/metadata", {
+        const {
+            categories,
+            dataElements,
+            indicators,
+            categoryOptions,
+            organisationUnits: organisationUnitsWithName,
+        } = await this.api.get("/metadata", {
             "categories:fields": "id,categoryOptions[id,code,name]",
-            "categories:filter": `code:in:[${categoryCode}]`,
+            "categories:filter": `code:in:[${dashboardItemsConfig.antigenCategoryCode}]`,
             "dataElements:fields": "id,code",
-            "dataElements:filter": `code:in:[${allDataElementCodes.join(",")}]`,
+            "dataElements:filter": `code:in:[${allDataElementCodes}]`,
             "indicators:fields": "id,code",
-            "indicators:filter": `code:in:[${allIndicatorCodes.join(",")}]`,
+            "indicators:filter": `code:in:[${allIndicatorCodes}]`,
+            "categoryOptions:fields": "id,categories[id,code],organisationUnits[id]",
+            "categoryOptions:filter": `organisationUnits.id:in:[${allAncestorsIds}]`,
+            "organisationUnits:fields": "id,displayName,path",
+            "organisationUnits:filter": `id:in:[${orgUnitsId}]`,
         });
-        const { id: categoryId, categoryOptions } = categories[0];
-        const antigensMeta = _.filter(categoryOptions, op => _.includes(antigenCodes, op.code));
+
+        const { id: antigenCategory } = categories[0];
+        const antigensMeta = _.filter(categories[0].categoryOptions, op =>
+            _.includes(antigenCodes, op.code)
+        );
+
+        if (!categoryOptions || !categoryOptions[0].categories) {
+            throw new Error("Organization Units chosen have no teams associated"); // TEMP: Check will be made dynamically on orgUnit selection step
+        }
+
+        const teamsByOrgUnit = organisationUnitsPathOnly.reduce((acc, ou) => {
+            let teams: string[] = [];
+            categoryOptions.forEach(
+                (co: { id: string; organisationUnits: object[]; categories: object[] }) => {
+                    const coIsTeam = _(co.categories)
```

Comment:
Added this by code. As we discussed, ideally we would add a filter to this.api.get, but D2 only supports arrays for specific keys ("filter", but no "model:filter" that we use in metadata endpoint). If, in the future, we create a wrapper over api.get, we can do using the filter.